src/lines_to_params.rs

```diff
@@ -0,0 +1,165 @@
+    let sequence = parser::u32_from_param(col.first().unwrap());
+    Python::with_gil(|py| {
+        let source_actor_obj = create_id_name_pair(py, "nari.types.actor", "Actor", source_actor);
+        let target_actor_obj = create_id_name_pair(py, "nari.types.actor", "Actor", target_actor);
+        update_position(source_actor_obj, source_position);
+        update_position(target_actor_obj, target_position);
+        update_resources(source_actor_obj, source_resources);
+        update_resources(target_actor_obj, target_resources);
+
+        let ability_event = create_id_name_pair(py, "nari.types.ability", "Ability", ability);
+
+        let action_effect = PyModule::import(py, "nari.types.actioneffect").unwrap()
+            .getattr("ActionEffect").unwrap();
+
+        let action_effects = action_effects
+            .chunks(2)
+            .map(|x| {
+                let (param0, param1, severity, effect_category, value, flags, multiplier) = action_effect_from_params(x.to_vec());
+                let kwargs = PyDict::new(py);
+                kwargs.set_item("effect_category", effect_category).unwrap();
+                kwargs.set_item("severity", severity).unwrap();
+                kwargs.set_item("flags", flags).unwrap();
+                kwargs.set_item("value", value).unwrap();
+                kwargs.set_item("multiplier", multiplier).unwrap();
+                kwargs.set_item("additional_params", (param0, param1)).unwrap();
+                action_effect.call((), Some(kwargs)).unwrap()
+            }).collect::<Vec<&PyAny>>();
+
+        let kwargs = PyDict::new(py);
+        kwargs.set_item("timestamp", timestamp).unwrap();
+        kwargs.set_item("action_effects", action_effects).unwrap();
+        kwargs.set_item("source_actor", source_actor_obj).unwrap();
+        kwargs.set_item("target_actor", target_actor_obj).unwrap();
+        kwargs.set_item("ability", ability_event).unwrap();
+        kwargs.set_item("sequence_id", sequence).unwrap();
+
+        PyModule::import(py, "nari.types.event.ability").unwrap()
+            .getattr("Ability").unwrap().call((), Some(kwargs)).unwrap().to_object(py)
+    })
+}
+
+/// Params to action_effect
+#[pyfunction]
+#[pyo3(text_signature = "(params: list[str]) -> list[int]")]
+pub(crate) fn action_effect_from_params(inp: Vec<&str>) -> ActionEffectParams {
+    let mut num = parser::u32_from_param(inp[0]);
+    let param0 = (&num >> 24) as u8;
+    let param1 = (&num >> 16) as u8;
+    let severity = (&num >> 8) as u8;
+    let effect_category = num as u8;
+    num = parser::u32_from_param(inp[1]);
+    let value = (&num >> 16) as u16;
+    let flags = (&num >> 8) as u8;
+    let multiplier = num as u8;
+    (param0, param1, severity, effect_category, value, flags, multiplier)
+}
+
+/// Params to status_effect
+#[pyfunction]
+#[pyo3(text_signature = "(params: list[str]) -> (int, int, float, int")]
+pub(crate) fn status_effect_from_params(inp: Vec<&str>) -> StatusEffectParams {
```

Comment:
Exactly the same here.

Reply:
Look above.

Reply:
Removed pyo3 export as this is internally used in another function